### PR TITLE
Added share button

### LINKED ISSUE #39: First iteration on Share + Detail View
It'll take a little time to develop good UX for the connections between project card, share button, detail view, and external links. 

Let's create a quick spec to so we can wrap this heartbeat or start the next with the sharing feature enabled and behaving "good enough".

 - add a Share button that reveals URL
 - change or remove trigger on detail view
 - clean up content in detail view

This might mean moving the "open" button to the detail view. Or, only displaying the detail view when viewed via link. 

Or – attempts at an easy fix might be too hard to justify. I hope not, but might be true.

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #39 

@acabunoc / @cadecairos review plz 🌹 

Note that I can't figure out how to get the direct link preselected on mobile. When this PR gets merged I'll create a follow-up ticket for that.

### CODE REVIEW

File: js/utility.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/* utility */
+
+var utility = {
+  findContainer : function(elem, containerClass){
+      while ((elem = elem.parentElement) && !elem.classList.contains(containerClass));
+      return elem;
+  },
+  getProjectIdFromUrl: function(url) {
```
Comment (mmmavis) [inline js/utility.js]: I can't decide if these helper methods should be part of `PulseMaker` or not

Comment (mmmavis) [inline js/utility.js]: Also... these are just quick url resolutions. In our next iteration these can be easily solved by external module / library.

Comment (acabunoc) [inline js/utility.js]: I think it's fine here :)

**Comment by mmmavis** [comment]
@acabunoc yup that's intended. Users cannot trigger the detail view from UI now but only through hitting the direct link (the url shown in the text box once you hit the "share" button)

**Comment by mmmavis** [comment]
Gonna merge this now! Yeaaaa thanks @acabunoc

**Pull request merged**